src/Codeception/Step.php

```diff
@@ -244,7 +244,11 @@ protected function humanize($text)
         $text = preg_replace('/([A-Z]+)([A-Z][a-z])/', '\\1 \\2', $text);
         $text = preg_replace('/([a-z\d])([A-Z])/', '\\1 \\2', $text);
         $text = preg_replace('~\bdont\b~', 'don\'t', $text);
-        return strtolower($text);
+        if (!preg_match('!!u', $text)) {
```

Comment:
This is an empty pattern and it matches any string I pass to it, so this code block is equivalent to unconditional `return mb_strtolower($text);`

Reply:
It matches any utf8 string. To get a negative result you need to give it non utf8 string.

Reply:
OK, confirmed.
But Codeception only supports UTF-8 (it is hardcoded it many places):
https://github.com/Codeception/Codeception/blob/c50789a9a62cc0eefc0252e88a5f04f8c47b55f4/src/Codeception/Lib/Console/Message.php#L110-L115
https://github.com/Codeception/Codeception/blob/c50789a9a62cc0eefc0252e88a5f04f8c47b55f4/src/Codeception/Scenario.php#L117-L128
https://github.com/Codeception/Codeception/blob/c50789a9a62cc0eefc0252e88a5f04f8c47b55f4/src/Codeception/Actor.php#L38
https://github.com/Codeception/Codeception/blob/c50789a9a62cc0eefc0252e88a5f04f8c47b55f4/src/Codeception/Step.php#L126

So unconditional mb_strtolower with hardcoded 'utf-8' charset would be better, unless you want to update all other places too.

Reply:
Yes, sure, I just though that non mb lowercase function could be used there for a reason so I was trying to do the change in a backward compatible way.